src/java/org/apache/cassandra/service/StorageServiceMBean.java

```diff
@@ -1124,16 +1124,6 @@ public void enableAuditLog(String loggerName, String includedKeyspaces, String e
     public boolean autoOptimisePreviewRepairStreams();
     public void setAutoOptimisePreviewRepairStreams(boolean enabled);
 
-    // warning thresholds will be replaced by equivalent guardrails
```

Comment:
I think we should keep the JMX methods, since it's a public API. We could change their implementation in `StorageService` to use the getters and setters in `Guardrails.instance` instead of using `DatabaseDescriptor`.

Reply:
Good point - I'll try doing that

Reply:
I also think they should be left `@Deprecated`, since they are only for backward compatibility. If we directly translate the arguments of those methods to the guardrail thresholds we are changing their meaning, since they used to count system keyspace/tables. 

I think the setter methods on `StorageService` should subtract the number of system keyspaces/tables. Conversely,  the getters should add the number of system keyspaces/tables to the guardrail number. That way the MBean methods would behave the same way they used to, being a JMX version of the old config properties.

Reply:
Hmm, yes, I think you are right about that. I was thinking that legacy values from the config would have been converted already, and users would want to work with current guardrail values. But, as you point out, that changes the semantics of the api, and since we are leaving the api methods deprecated they should work the same.